chore: :label: Adding Types to Medusa Configuration

adevinwild opened the pull request:
Added a JSDoc `@type` to the `medusa-config.js` file.

Based on this discussion : https://github.com/medusajs/medusa/discussions/3275

adevinwild committed:
chore: :label: Adding Types to Medusa Configuration

Added a JSDoc `@type` to the `medusa-config.js` file.

Based on this discussion : https://github.com/medusajs/medusa/discussions/3275

olivermrbl reviewed (changes requested):
This is great @adevinwild! 

Can I get you to import from `/** @type {import('@medusajs/medusa').ConfigModule} */` instead of `dist`? 😊

adevinwild committed:
chore: :label: Better export path

Exporting directly from the `@medusajs/medusa` package
Based on this discussion : https://github.com/medusajs/medusa/discussions/3275

adevinwild commented:
@olivermrbl  It's fixed, sorry I had a problem and I was trying to understand why in local I didn't have the different types exported and so I used this path haha !

olivermrbl commented:
@adevinwild Thanks for the contribution! 🥳
